File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s;
        for (int i: nums){
            s.insert(i);
        }
        int best = 0;
        for (int i: nums){
            if (s.contains(i)){
                int left = 0, right = 0;
                int j = i+1;
                while (s.contains(j)){
                    s.erase(j);
                    j++;
                    right++;
                }
                j=i-1;
                while(s.contains(j)){
                    s.erase(j);
                    j--;
                    left++;
                }
                best = max(best,left+right+1);
            }
        }
        return best;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        best = 0
        i = 0
        j = len(heights)-1
        while (i<j):
            best = max(best,(j-i)*min(heights[i],heights[j]))
            if heights[i]<heights[j]:
                i+=1
            else:
                j-=1
        return best